_src/modules/card.js

```diff
@@ -0,0 +1,249 @@
+import $ from 'jquery';
+import { Liquid } from 'liquidjs';
+import tree_func from '../tools/tree';
+import { formatDate } from '../tools/date';
+import Pluralization from '../tools/pluralize';
+import { getLayoutVars } from './layout_vars';
+import accordion from './accordion';
+
+const engine = new Liquid();
+
+const isDefaultLayout = getLayoutVars().layout === 'default';
+const isSF = getLayoutVars().type === 'sf';
+const dirUrl = isSF ? 'person_sf/' : 'person/';
+const feedbackForm = isSF
+    ? 'https://docs.google.com/forms/d/e/1FAIpQLScWdkT2GB6btR3duE5fWRzbTeg8HeLji8MR7ZW2-0oMiG7-Wg/' 
+    : 'https://docs.google.com/forms/d/e/1FAIpQLSd6WGVWRwdwf7Q7oE_3RIlHR8MMdo_LVOnC0nk9YINtfYvjPw/';
+
+let currentCardData = null;
+let currentLawSelected = null;
+let currentExpandedAccordions = new Set();
+let currentScrollTop = 0;
+let template;
+
+let cardNode;
+
+if (isDefaultLayout) {
+    
+    cardNode = $('#card');
+    const templateHtml = cardNode.find("#card_template").html();
+    template = engine.parse(templateHtml);
+
+    cardNode.html('');
+
+    cardNode.on('click', '#scrollToTop', function (event) {
+        event.preventDefault();
+        cardNode.animate({
+            scrollTop: 0
+        }, 500);
+    });
+
+    cardNode.on('scroll', function() {
+        if (cardNode.scrollTop() > 200 ) {
+            cardNode.find('#scrollToTop').show();
+        } else {
+            cardNode.find('#scrollToTop').hide();
+        }
+    });
+
+    cardNode.on(accordion.EVENT_TOGGLE, '.card__main-info .accordion', function(event, isOpen) {
+        const id = $(this).attr('data-id');
+        if (isOpen) {
+            currentExpandedAccordions.add(id);
+        } else {
+            currentExpandedAccordions.delete(id);
+        }
+    });
+
+    cardNode.on('click', '.card__law-details-button', function (event) {
+        event.preventDefault();
+        const lawId = $(this).closest('.card__law').attr('data-id');
+        showLawDetails(lawId);
+    });
+
+    cardNode.on('click', '.card__law-back-button', function (event) {
+        event.preventDefault();
+        hideLawDetails();
+    });
+
+    cardNode.on('click', '#close_btn', HideCard);
+    $(window).on('popstate', HideCard); // back button should close modal
+}
+
+function HideCard() {
+    cardNode.hide();
+    cardNode.html('');
+
+    currentCardData = null;
+    currentLawSelected = null;
+    currentScrollTop = 0;
+    currentExpandedAccordions.clear();
+
+    window.dispatchEvent(new Event('closeCard'));
+    if (window.history && window.history.pushState) {
+        window.history.pushState('backward', null, isSF ? './sf' :'./');
+    }
+}
+
+function ShowCard(depInfo, depRating, depLobbys, lobby_list, declarations) {
```

Comment:
кейс: открываем карточку, а затем открываем другую. предложение: убеждаемся что скролл вверху, аккордионы закрыты
```suggestion
function ShowCard(depInfo, depRating, depLobbys, lobby_list, declarations) {
HideCard();
```

Reply:
В этом нет необходимости. В функции ShowCard оба кейса решаются.

Reply:
не решаются. и аккордионы не закрывается и скролл к самому верху не скроллится. 
![image](https://user-images.githubusercontent.com/10209901/216286713-b3b34ed7-15ed-4395-ba91-f4becf5aa4ba.png)